### PR TITLE
Feature/trainers section

### PR DESCRIPTION
**Pull request by AarCon**
## Summary
This PR does a couple things. The main thing is that it adds the Trainers Section to the Mapper. It also changes a few minor details in components like the PokemonAccordion, PokemonMovesetList, pokemonStatCalculation and the SearchBar. This is still a WIP only because I need to change out the trainer files to be more current.

### Mapper
Adds the TrainersSection and initializes the trainerList when a zone is selected. Settings Button was moved to be inside of the "SearchBar". The position will be changed to be in the bottom right corner of the map.

### SearchBar
As mentioned before, the Settings Button was moved here to be better positioned on the map.

image

### PokemonAccordion
Changed this component to be controlled and allowed parent components to dictate the open or closed state. This is important for the Trainer Selection to open the accordion when a team is selected and close when the field is cleared.

image

Additionally, the `sx` prop can be accessed from the parent to adjust any css properties on an individual level.

### PokemonStatCalculation
This function was adjusted to get more accurate results when it comes to stat calculation. Nature multipliers were added to this function to condense the stat calculating to one place. There was also a bug when it came to HP that had an extra 100 being added erroneously.

### PokemonStats
Now, this isn't technically a new component but it was different enough from the original that is was necessary to separate them. The main addition is the trainer pokemon addition. This will now take in all of the values associated with a trainer pokemon and display it in a neat little rectangle.

image

But not only that, it has two modes, the Extended Info mode as seen above and the Condensed Info mode mainly for use with mobile.

image

### Trainers
This is where a lot of the magic happens for visualizing the trainer teams. This took many hours of going back and forth between design concepts and real life application, but we finally have a solid component that is also responsive. Additionally, much of this is reused code from the pokedex. You may recognize the move type icons that are used for the pokemon types or the stat bars from the original PokemonStats component.

The only real complicated thing here is the styling to allow it to be used with mobile. The way this was achieved is to go back and forth between Condensed and Expanded Info modes. The order (from smallest screen size to largest) is: 1 mon condensed => 1 mon expanded => 2 mons condensed => 2 mons expanded. You can also see that below.

https://github.com/TeamLumi/luminescent-team-dev/assets/125856819/b329f853-3fcb-47a4-b27e-4c45cea16229

### TrainersSection
This component handles much of the state management for all of the Trainer components and makes sure that everything is in step with each other. One way that it does this is by clearing out the trainer after a new zone has been selected. When the trainer is cleared, then the PokemonAccordion is closed to avoid any errors with the Trainer Dropdown

### CODE REVIEW

**Review by mKroogz — APPROVED** [review]
Looks amazing. :)

**Comment by AarCon** [comment]
Alright I've switched up a bunch of things because I wasn't happy with how the Trainers Section looked at the bottom of the page

## Summary 
I added buttons to the top of the encounters side panel to make it a better UX especially for mobile users (whenever that can come to fruition). For the main pages in the side panel, it will have Encounters, Trainers, Items and Shops. Due to the limiting width of the side panel however, I made a Trainers Modal that you can open when a trainer is selected. The last major change is that I refactored a lot of files to organize them in separate folders.

https://github.com/TeamLumi/luminescent-team-dev/assets/125856819/68836df2-f6d1-4b06-8c9d-537e4d5a5943

## Changes to Existing Code (since last review)
### PokemonStats
Added a `smallest` prop that will force the styles to be at their smallest size if true. This is used specifically for the Trainers Side Panel. In conjunction with that, a function was made that takes a css style prop and returns the `smallest` style if it's true.

### Mapper
Adds the TrainersModal to show off a more detailed view of Trainers. Introduces a tab panel to make for a better User Experience. Changes the location name being passed to the `getTrainersFromZoneName` to be the zoneId for more accurate results.

### Trainers
Moved this file into a folder that makes more sense. Fractured this into more components for easier troubleshooting and general organization. Changed the `MonDetails` Ability to use the existing `PokemonAbility` that opens a modal when clicked. Changed the `MoveList` component to use `PokemonMove`  which turns the move into a button that opens a modal with more details about that specific move.

### Coordinates
I noticed that we didn't have the Distortion Room (where we fight Cyrus and Giratina) so I moved some locations around to fit it in. Also changed the zoneId of the Pokemon League to match with the `Gamedata` repo's generated data. 
<details>
  <summary>New arrangement of Mt. Coronet to include Distortion Room</summary>

image

And here's the key
```python
zones = [
    211, # Mt. Coronet 6F
    216, # Spear Pillar
    624, # Distortion Room
    218, # Hall of Origin
    205, # Mt. Coronet 3F
    204, # Mt. Coronet 2F
    203, # Mt. Coronet Route 207
    213, # Mt. Coronet 216 Entrance
    215, # Mt. Coronet B1F
    214, # Mt. Coronet 211 Entrance
    212, # Mt. Coronet Tunnel To 211 Entrance
    208, # Mt. Coronet 4F
    207, # Mt. Coronet Summit
    210, # Mt. Coronet 5F
    209, # Mt. Coronet 4F To Spear Pillar
    206, # Mt. Coronet Snow Area
    999, # Regice Ruins
]
```

</details>


### Pokemon Ability
Added a prop to include `Ability: ` as a label. Also fixed some linting and pass-through props.

### Pokemon Moveset
Refactored the `PokemonMoveType` component to better capture styling necessities. Added the `PokemonMove` component which transforms the `PokemonMoveType` component into a button and opens a Modal which displays the `MovesetListItem`  for a specific move.

## New Components
### Pokemon Tabs
This is a custom component that we will likely use in the future whenever the need arises. It basically takes MUI's existing `Tabs` and `TabPanel` components and makes them more in line with how we would actually want to use them.

### Mapper Tab Panel
This will take all of the different Panels and organize them how we want them in the main `Mapper` component.

### Trainer Tab Panel
This displays the Trainers Section in the smallest style it can be initially. There is also a button called "Detailed" (Still not set on the name of it tbh) that opens the `TrainersModal` in as large as the current display can handle. This button will not be present in the smallest screen size.

### Trainers Modal
This displays the `Trainers` component in a MUI dialog. Not too much to say on this component lol.

File: src/components/Mapper/Encounters/EncounterTable.jsx [hunk 1]
```diff
@@ -11,9 +11,9 @@ import {
   Typography
 } from '@mui/material';
 
-import "./style.css";
+import ".././style.css";
```
Comment (mKroogz) [inline src/components/Mapper/Encounters/EncounterTable.jsx]: That's interesting that it made ya change that 🤔

File: src/components/Mapper/Encounters/EncounterTable.jsx [hunk 2]
```diff
@@ -11,9 +11,9 @@ import {
...
 
-export const EncounterTable = ({ encounterList, pokemon }) => {
+const EncounterTable = ({ encounterList, pokemon }) => {
```
Comment (mKroogz) [inline src/components/Mapper/Encounters/EncounterTable.jsx:16]: Cool stuff 😎

File: src/components/Mapper/Encounters/PokemonStats.jsx
Comment (mKroogz) [inline src/components/Mapper/Encounters/PokemonStats.jsx:1]: This is awesome!!!

File: src/components/Mapper/Mapper.jsx [hunk 3]
```diff
@@ -10,15 +10,15 @@ import {
...
-import SettingsModal from './SettingsModal';
+import { RodButtons, TimeOfDayButtons } from './Encounters/Buttons';
+import { MapperTabPanel } from './TabPanel/MapperTabPanel';
+import SettingsModal from './Settings/SettingsModal';
 import './style.css';
 
 import {
   getAreaEncounters,
-  getTrainersFromZoneName,
+  getTrainersFromZoneId,
```
Comment (mKroogz) [inline src/components/Mapper/Mapper.jsx:21]: Good changes with id vs name

File: src/components/Mapper/Settings/HighlightColors.jsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { TextField, Typography, Grid, Button } from '@mui/material';
-import './style.css';
+import '../style.css';
```
Comment (mKroogz) [inline src/components/Mapper/Settings/HighlightColors.jsx:3]: Ah wait I see the rework now nice.

File: src/components/Mapper/TabPanel/ItemsPanel.jsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { Box } from "@mui/material";
+
+const ItemsPanel = () => {
+  return (
+    <Box>
+      Coming Soon.
```
Comment (mKroogz) [inline src/components/Mapper/TabPanel/ItemsPanel.jsx:7]: 😎

File: src/components/common/ImageWithFallback.jsx [hunk 6]
```diff
@@ -1,9 +1,14 @@
 import useBaseUrl from '@docusaurus/useBaseUrl';
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 
 export const ImageWithFallback = ({ fallbackSrc, src, ...props }) => {
   const [imgSrc, setImgSrc] = useState(src);
+
+  useEffect(() => {
```
Comment (mKroogz) [inline src/components/common/ImageWithFallback.jsx:7]: Love the new use effect for image src management that's really cool.

File: src/core/pokemonStatCalculation.js
Comment (mKroogz) [inline src/core/pokemonStatCalculation.js:1]: Fully fledged and mathed out stats nice 😎

**Review by mKroogz — APPROVED** [review]
This all looks awesome 😁

**Pull request merged**